### PR TITLE
Filter out packages with dependencies that have open PRs in rebuild

### PR DESCRIPTION
**Pull request by justcalamari**

### CODE REVIEW

File: conda_forge_tick/migrators.xsh [hunk 1]
```diff
@@ -536,11 +536,11 @@ class Rebuild(Migrator):
                 f.write(upd)
 
     def filter(self, attrs):
-        if attrs.get('archived', False) or attrs.get('bad', False):
+        if super().filter():
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: I think super filter takes attrs?

File: conda_forge_tick/migrators.xsh [hunk 2]
```diff
@@ -536,11 +536,11 @@ class Rebuild(Migrator):
...
             return True
         for node in self.graph.predecessors(attrs['feedstock_name']):
             att = self.graph.node[node]
-            if self.migrator_uid(att) in att.get('PRed', []):
+            if self.migrator_uid(att) not in att.get('PRed', []) or att.get('PRed_json', {}).get('state', '') == 'open':
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: would it be possible to break this up over multiple lines?

Comment (CJ-Wright) [reply conda_forge_tick/migrators.xsh]: Also is `PRed_json` a dict with `migrator_uid` keys?

**Comment by justcalamari** [comment]
Ready for review @CJ-Wright

File: conda_forge_tick/migrators.xsh [hunk 3]
```diff
@@ -508,9 +508,10 @@ class Rebuild(Migrator):
                        '{{% set build = {} %}}')
                      )
 
-    def __init__(self, graph, pr_limit=0):
+    def __init__(self, graph, name, pr_limit=0):
```
Comment (scopatz) [inline conda_forge_tick/migrators.xsh]: I think it you are going to change the API for one migrator in a required way, you should change it for all migrators.  Like perhaps this should be on the `Migrator` class or a keyword argument.

Comment (CJ-Wright) [reply conda_forge_tick/migrators.xsh]: :+1: for having name as kwarg

File: conda_forge_tick/migrators.xsh [hunk 4]
```diff
@@ -508,9 +508,13 @@ class Rebuild(Migrator):
                        '{{% set build = {} %}}')
                      )
 
-    def __init__(self, graph, pr_limit=0):
+    def __init__(self, graph=None, name=None, pr_limit=0):
         super().__init__(pr_limit)
-        self.graph = graph
+        if graph == None:
+            self.graph = nx.DiGraph()
```
Comment (CJ-Wright) [inline conda_forge_tick/migrators.xsh]: Clever, I like it, it turns it into a no-op if no graph given.

**Comment by justcalamari** [comment]
We can either add the python3.7 migration to auto_tick in this PR or merge this and put that in a separate PR. Let me know what you think, @CJ-Wright

**Comment by CJ-Wright** [comment]
Separate PR please!

**Pull request merged**